store/rootmulti/store.go

```diff
@@ -294,22 +292,59 @@ func (rs *Store) LastCommitID() types.CommitID {
 
 // Commit implements Committer/CommitStore.
 func (rs *Store) Commit() types.CommitID {
-
-	// Commit stores.
-	version := rs.lastCommitInfo.Version + 1
+	previousHeight := rs.lastCommitInfo.Version
+	version := previousHeight + 1
 	rs.lastCommitInfo = commitStores(version, rs.stores)
 
-	// write CommitInfo to disk only if this version was flushed to disk
-	if rs.pruningOpts.FlushVersion(version) {
-		flushCommitInfo(rs.db, version, rs.lastCommitInfo)
+	// Determine if the current height needs to be added to the list of heights to
```

Comment:
incorrect docs: We are not determing if current height needs to be pruned here, we're seeing if pruneHeight should be pruned

`pruneHeight != currentHeight`